### DISCUSSION TITLE
Question about cation conservation in the entire domain.

### DISCUSSION THREAD

**Discussion by TangLeiPlus** [pybamm-team/PyBaMM#2426] [Q&A] [answered]
Hi everyone
I have a question about the total number of cations in a battery cell.
I tried to calculate the total number of cation in Chen2020 parameter set. I caculate it in two different ways which are
Approach 1:
positive electrode volume * positive electrode concentration + electrolyte volume * electro concentration + negative electrode volume * negative electride concentration = 0.289 mol
(the concentrations here are all initial value in the parameter set)
Approach 2:
positive electrode volume * positive electrode density / positive electrode molar mass = 0.3258 mol

0.289 mol < 0.3258 mol
where do the other 0.033mol catations go? Is there any one can help me to understand this?
Many Thanks

**Comment by RuiheLi** [comment]
I think that is probably because electrode density has some errors. From my personal experience this is a parameter that not very much important in the DFN model, at least it seldom occurs in equations.

**Comment by valentinsulzer** [comment, marked as answer]
You can see how the total number of moles should be calculated here https://github.com/pybamm-team/PyBaMM/blob/959081ee329f49cb6f6fc04a4f420da056500a02/pybamm/parameters/lithium_ion_parameters.py#L139 and https://github.com/pybamm-team/PyBaMM/blob/959081ee329f49cb6f6fc04a4f420da056500a02/pybamm/parameters/lithium_ion_parameters.py#L783
